classlib/src/main/java/org/teavm/classlib/java/lang/TDouble.java

```diff
@@ -201,7 +202,9 @@ public boolean equals(Object other) {
         if (this == other) {
             return true;
         }
-        return other instanceof TDouble && doubleToLongBits(((TDouble) other).value) == doubleToLongBits(value);
+        return other instanceof TDouble && PlatformDetector.isJavaScript()
+                ? TDouble.compare(((TDouble) other).value, value) == 0
```

Comment:
Instead of comparison it's also possible to do following:

```java
a == b && (1 / a) == (1 / b) || a != a && b != b
```

which I believe a bit faster